src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py

```diff
@@ -2,98 +2,120 @@
+
+
+def _clone_and_replace_action_group(cmd, source_resource, alert_rule, action_group_mapping, target_resource):
+    source_subscription_id = parse_resource_id(source_resource)['subscription']
+    target_subscription_id = parse_resource_id(target_resource)['subscription']
+    for index, action in enumerate(alert_rule['actions']):
+        if action['action_group_id'] in action_group_mapping:
+            alert_rule['actions'][index] = action_group_mapping[action['action_group_id']][1]
         else:
-            resource_group_name, name = _parse_id(action.action_group_id).values()  # pylint: disable=unbalanced-dict-unpacking
-            action_group = source_monitor_client.action_groups.get(resource_group_name, name)
+            resource_group_name, name = _parse_id(action["action_group_id"]).values()  # pylint: disable=unbalanced-dict-unpacking
+            from ..aaz.latest.monitor.action_group import Show
+            action_group = Show(cli_ctx=cmd.cli_ctx)(command_args={
+                'subscription': source_subscription_id,
+                "resource_group": resource_group_name,
+                "action_group_name": name,
+            })
+
+            from .action_groups import ActionGroupCreate
             name = CLONED_NAME.format(name, gen_guid())
             resource_group_name, _ = _parse_id(target_resource).values()  # pylint: disable=unbalanced-dict-unpacking
-            new_action_group = target_monitor_client.action_groups.create_or_update(resource_group_name,
-                                                                                    name,
-                                                                                    action_group)
-            MetricAlertAction = cmd.get_models('MetricAlertAction', operation_group='metric_alerts')
-            new_action = MetricAlertAction(action_group_id=new_action_group.id,
-                                           web_hook_properties=action.web_hook_properties)
-            alert_rule.actions[index] = new_action
-            action_group_mapping[action.action_group_id] = [new_action_group.id, new_action]
+            action_group["subscription"] = target_subscription_id
+            action_group["resource_group"] = resource_group_name
+            action_group["action_group_name"] = name
+            new_action_group = ActionGroupCreate(cli_ctx=cmd.cli_ctx)(command_args=action_group)
+            new_action = {
+                "action_group_id": new_action_group["id"],
+                "web_hook_properties": action.get("web_hook_properties", {})
+            }
+            alert_rule['actions'][index] = new_action
+            action_group_mapping[action['action_group_id']] = [new_action_group['id'], new_action]
     return alert_rule
 
 
-def _clone_alert_rule(monitor_client, alert_rule, target_resource):
-    alert_rule.scopes = [target_resource]
+def format_metrics_alert_req(alert_rule):
+    all_of = alert_rule["criteria"]["all_of"]
```

Comment:
Do you need to pop the "all_of"?

Reply:
`all_of` for `criteria` will be override in the following operations.